security/cmd/node_agent/server.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodeagentmgmt
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/gogo/googleapis/google/rpc"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/security/cmd/node_agent/na"
+	"istio.io/istio/security/cmd/node_agent_k8s/workload/handler"
+	cagrpc "istio.io/istio/security/pkg/caclient/grpc"
+	pkiutil "istio.io/istio/security/pkg/pki/util"
+	"istio.io/istio/security/pkg/platform"
+	"istio.io/istio/security/pkg/workload"
+	pb "istio.io/istio/security/proto"
+)
+
+// Server specifies the node agent server.
+// TODO(incfly): adds sync.Mutex to protects the fields, such as `handlerMap`.
+type Server struct {
+	// map from uid to workload handler instance.
+	handlerMap map[string]handler.WorkloadHandler
+	// makes mgmt-api server to stop
+	done     chan bool
+	config   *na.Config
+	pc       platform.Client
+	cAClient cagrpc.CAGrpcClient
+	// the workload identity running together with the NodeAgent, only used for vm mode.
+	identity string // nolint
+	// ss manages the secrets associated for different workload.
+	ss workload.SecretServer
+}
+
+// determineEnv chooses the right platform. If the env is specified in cfg.Env,
+// then we will use it. Otherwise nodeagent will detect the platform for you.
+func determineEnv(cfg *na.Config) string {
```

Comment:
This func is used in other file as well, consider to make a util func and reuse? A TODO is fine for this pr

Reply:
My plan is to merge nodeagent.go and node_agent/management/server.go in a single file later.

Changing this func to public temporarily to avoid code duplicating.